app/src/main/java/com/internship/move/customviews/LoadingButton.kt

```diff
@@ -0,0 +1,52 @@
+package com.internship.move.customviews
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.FrameLayout
+import com.internship.move.R
+import com.internship.move.databinding.ViewLoadingButtonBinding
+
+class LoadingButton @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null
+) : FrameLayout(context, attrs) {
+
+    private val binding = ViewLoadingButtonBinding.inflate(LayoutInflater.from(context), this)
+    private var btnText: CharSequence = ""
+    var isLoading: Boolean = false
+        set(value) {
+            field = value
+            if (value) {
+                binding.loadingPB.visibility = View.VISIBLE
+                binding.loadingBtn.text = ""
+                binding.loadingBtn.isClickable = false
+            } else {
+                binding.loadingPB.visibility = View.INVISIBLE
+                binding.loadingBtn.text = btnText
+                binding.loadingBtn.isClickable = true
+            }
+        }
+
+    init {
+        val a = context.obtainStyledAttributes(attrs, R.styleable.LoadingButton)
+        binding.loadingBtn.text = a.getText(R.styleable.LoadingButton_text)
```

Comment:
You're setting the button text here and under the `recycle` call. I think one set would be enough.
Also, you're getting the styleable text value twice, you can get it only once in `btnText` and use that